src/main/java/com/bobocode/controller/UserController.java

```diff
@@ -0,0 +1,84 @@
+package com.bobocode.controller;
+
+import com.bobocode.dto.Note;
+import com.bobocode.dto.User;
+import com.bobocode.mapper.NoteRepresentationMapper;
+import com.bobocode.mapper.UserRepresentationMapper;
+import com.bobocode.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserService userService;
+    private final UserRepresentationMapper userRepresentationMapper;
+    private final NoteRepresentationMapper noteRepresentationMapper;
+
+    @GetMapping("/{id}")
+    public ResponseEntity<EntityModel<User>> getUserById(@PathVariable String id) {
+        return ResponseEntity.ok(userRepresentationMapper.toModel(userService.getUserById(id)));
+    }
+
+    @GetMapping
+    public ResponseEntity<CollectionModel<EntityModel<User>>> findAll() {
+        return ResponseEntity.ok(userRepresentationMapper.toCollectionModel(userService.findAll()));
+    }
+
+    @PostMapping
+    public ResponseEntity<?> createUser(@RequestBody User user) {
+        userService.createUser(user);
+        URI location = ServletUriComponentsBuilder.fromCurrentRequestUri()
+                .path("/{id}")
+                .buildAndExpand(user.getId())
+                .toUri();
+        return ResponseEntity.created(location).build();
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void deleteUserById(@PathVariable String id) {
+        userService.deleteUserById(id);
+    }
+
+    @PostMapping("/{id}/notes")
```

Comment:
I would name user id as `{userId}` not just `{id}` as you've done here `/{userId}/notes/{noteId}` for consistency

Reply:
Thanks!
Actually, I had a doubt to name it as {userId} here, since it was obvious to me that this is id of the user after '/user'.
The reason why I called it {userId} in the 'notes' API because I couldn't had two {id} in the path.

But I'll stick to your advice, it will be more consistent in that way.

Reply:
Fixed